src/dmd/statement.d

```diff
@@ -1603,6 +1603,10 @@ extern (C++) final class SwitchStatement : Statement
             {
```

Comment:
```
dmd/statement.d(1600): Error: function dmd.statement.SwitchStatement.checkLabel.checkVar no return exp; or assert(0); at end of function
posix.mak:450: recipe for target '../generated/freebsd/release/32/dmd_frontend' failed
```

I think this is a bug in the host compiler. I guess combining the `if`s to a joint expression could be worth a try:

```d
if (last == vd ||
    vd.storage_class & STC.temp)
``